### PR TITLE
Api: fix validations for presign operations.

### LINKED ISSUE #170: Presigned Get and Put need to enforce default timeout and support response headers
- Need to raise an error if time to expire > 7 days. If missing expiry duration, need to set expiry default to 7 days
- Add support for additional response header overrides to bring in conformity with other SDKs

- Needs functional tests Refer #156 
 16. missing test for presignedgetobject() with default timeout
 17. missing test for presignedgetobject() with response headers.
 18. missing test for presignedputobject() with default timeout.

### PR DESCRIPTION
**Pull request by poornas**
Fixes #170  and the missing presigned Get/Put functional tests for Mint in #156 

Add validation of expiry for presign get and put.Add optional override response
headers as argument for PresignedGetObjectAsync method.

### CODE REVIEW

**Comment by poornas** [comment]
@donatello , pls review

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 1]
```diff
@@ -118,7 +118,7 @@ public static void Main(string[] args)
 
             // Set app Info 
             minioClient.SetAppInfo("app-name", "app-version");
-
+
```
Comment (donatello) [inline Minio.Functional.Tests/FunctionalTest.cs]: stray spaces

**Comment by deekoder** [comment]
waiting on travis failure fix?

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 1 above)
Comment (poornas) [reply Minio.Functional.Tests/FunctionalTest.cs]: fixed.

**Comment by poornas** [comment]
travis is currently not working with this distribution. of linux. So this failure is expected.

**Comment by harshavardhana** [comment]
> travis is currently not working with this distribution. of linux. So this failure is expected.

Yeah `travis` is set as `not-required` until this is fixed.

**Pull request merged**